### PR TITLE
Adds support for Rails' new error handling API

### PR DESCRIPTION
**Pull request by nikz**
If config.register_rails_error_handler is truthy, Raygun registers for errors via `Rails.error.handle`

See https://edgeguides.rubyonrails.org/error_reporting.html

### CODE REVIEW

File: lib/raygun/railtie.rb [hunk 1]
```diff
@@ -31,9 +31,17 @@ class Raygun::Railtie < Rails::Railtie
...
+
+    Raygun::Railtie.setup_error_subscriber
   end
 
   rake_tasks do
     load "tasks/raygun.tasks"
   end
+
+  def self.setup_error_subscriber
+    if ::Rails.version.to_f >= 7.0 && Raygun.configuration.register_rails_error_handler
```
Comment (sumitramanga) [inline lib/raygun/railtie.rb:43]: Do we need a fallback if this isn't met?

Comment (nikz) [reply lib/raygun/railtie.rb:43]: Not in this case as this is a new API, so exceptions will still be reported via the middleware.

Comment (sumitramanga) [reply lib/raygun/railtie.rb:43]: Sounds good, TY!

**Review by sumitramanga — APPROVED** [review]
Code LGTM! I'd be keen to test this out before a full release goes out 🚀

**Review by sumitramanga — CHANGES_REQUESTED** [review]
I see a few tests failing so it would be good to fix those bad boys up first :)

File: test/unit/sidekiq_failure_test.rb [hunk 2]
```diff
@@ -16,17 +16,20 @@ class SidekiqFailureTest < Raygun::UnitTest
...
     super
+    Raygun.configuration.send_in_background = false
 
     stub_request(:post, 'https://api.raygun.com/entries').to_return(status: 202)
     fake_successful_entry
   end
 
   def test_failure_backend_appears_to_work
-    assert Raygun::SidekiqReporter.call(
+    response = Raygun::SidekiqReporter.call(
```
Comment (sumitramanga) [inline test/unit/sidekiq_failure_test.rb:26]: These tests read much easier now!

**Comment by sumitramanga** [comment]
I'm happy with these changes :)

**Pull request merged**